sycl/unittests/pi/level-zero/test_ext_command_buffers.cpp

```diff
@@ -0,0 +1,331 @@
+//==---- test_ext_command_buffers.cpp --- PI unit tests
+//---------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "spirv-source.hpp"
+
+#include "gtest/gtest.h"
+
+#include "TestGetPlugin.hpp"
+#include <detail/plugin.hpp>
+#include <pi_level_zero.hpp>
+#include <sycl/detail/pi.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+class LevelZeroCommandBuffersTest : public ::testing::Test {
+protected:
+  std::optional<detail::plugin> Plugin =
+      pi::initializeAndGet(backend::ext_oneapi_level_zero);
+  pi_context Context = nullptr;
+  pi_device Device = nullptr;
+  pi_queue Queue = nullptr;
+
+  void SetUp() override {
+    // skip the tests if the Level Zero backend is not available
+    if (!Plugin.has_value()) {
+      GTEST_SKIP();
+    }
+
+    pi_uint32 NumPlatforms = 0;
+    pi_platform Platform = nullptr;
+
+    ASSERT_EQ(Plugin->getBackend(), backend::level_zero);
+
+    ASSERT_EQ((Plugin->call_nocheck<detail::PiApiKind::piPlatformsGet>(
+                  0, nullptr, &NumPlatforms)),
+              PI_SUCCESS)
+        << "piPlatformsGet failed.\n";
+
+    ASSERT_EQ((Plugin->call_nocheck<detail::PiApiKind::piPlatformsGet>(
+                  NumPlatforms, &Platform, nullptr)),
+              PI_SUCCESS)
+        << "piPlatformsGet failed.\n";
+
+    ASSERT_GE(NumPlatforms, 1u);
+    ASSERT_NE(Platform, nullptr);
+
+    ASSERT_EQ((Plugin->call_nocheck<detail::PiApiKind::piDevicesGet>(
+                  Platform, PI_DEVICE_TYPE_GPU, 1, &Device, nullptr)),
+              PI_SUCCESS)
+        << "piDevicesGet failed.\n";
+
+    ASSERT_EQ((Plugin->call_nocheck<detail::PiApiKind::piContextCreate>(
+                  nullptr, 1, &Device, nullptr, nullptr, &Context)),
+              PI_SUCCESS)
+        << "piContextCreate failed.\n";
+
+    ASSERT_EQ((Plugin->call_nocheck<detail::PiApiKind::piQueueCreate>(
+                  Context, Device, 0, &Queue)),
+              PI_SUCCESS)
+        << "piContextCreate failed.\n";
+
+    EXPECT_NE(Context, nullptr);
+  }
+
+  LevelZeroCommandBuffersTest() = default;
+
+  ~LevelZeroCommandBuffersTest() = default;
+};
+
+TEST_F(LevelZeroCommandBuffersTest, piextCommandBufferCreate) {
+  // Create command buffer
+  pi_ext_command_buffer CommandBuffer = nullptr;
+  pi_ext_command_buffer_desc CommandBufferDesc;
+  ASSERT_EQ((Plugin->call_nocheck<detail::PiApiKind::piextCommandBufferCreate>(
+                Context, Device, &CommandBufferDesc, &CommandBuffer)),
+            PI_SUCCESS);
```

Comment:
Could have a check for `ASSERT_NE(CommandBuffer, nullptr);`, but could just do that in this test to avoid making other tests noisy

Reply:
Done!